src/globus_cli/commands/endpoint/server/show.py

```diff
@@ -1,14 +1,54 @@
 from __future__ import annotations
 
+import typing as t
 from textwrap import dedent
 
 from globus_cli.login_manager import LoginManager
 from globus_cli.parsing import command, endpoint_id_arg
-from globus_cli.termio import FORMAT_TEXT_RECORD, formatted_print
-from globus_cli.types import FIELD_LIST_T
+from globus_cli.termio import Field, TextMode, display, formatters
 
 from ._common import server_id_arg
 
+PORT_RANGE_T = t.Tuple[int | None, int | None]
```

Comment:
Looking at this again, I do not believe that `(int, None)` and `(None, int)` are valid.
I'll update this. My plan is to make this type alias `tuple[int, int] | None` and treat `None` as the `"unspecified"` case below.

Reply:
Sounds good!

Reply:
I believe https://github.com/globus/globus-cli/pull/692/commits/cb0438f7aa2d37dcc67552b003cb97efba152a78 handles this well (although it could stand to have some unit tests attached to it, which I think I'll do right now).